server/lib/enrich/article.js

```diff
@@ -31,6 +31,13 @@ module.exports = exports = function article(content, format) {
 		content.bodyHTML = content.body;
 	}
 
+	content.hasGraphics = content.contentStats && content.contentStats.graphics && content.contentStats.graphics > 0 ? true : false;
+
+	const countOfGraphics = content.contentStats && content.contentStats.graphics && content.contentStats.graphics;
+	const countOfSharableGraphics = content.embeds ? content.embeds.filter(embed => embed && embed.type.endsWith('Graphic')).filter(embed => embed.canBeSyndicated === 'yes').length : 0;
+
+	content.canAllGraphicsBeSyndicated = countOfGraphics > 0 ? countOfSharableGraphics < countOfGraphics : false;
+
```

Comment:
The `embeds` array contains all the images and graphics in all their different sizes. However, only the original (default size) images are used for syndication.

Should the graphics included in the content document be the source of truth for graphics in the article instead of the embeds array? 

the embeds array would then be used as a map to check whether the graphics can be syndicated or not.

Reply:
But that's what's there, with `content.hasGraphics` relying on `contentStats.graphics` count? (line 34)?

Then you have `countOfGraphics` (once again based on the `contentStats`). And `countOfSharableGraphics` which runs through the `embeds` to see how many can be syndicated. 

It feels like you know something I don't - I'm just using an example ES article to figure this out 😅

Reply:
contentStats.graphics is gotten from [counting the number of graphics](https://github.com/Financial-Times/next-es-interface/blob/master/server/transformers/plugins/article/content-stats.js#L25) in the content document. embeds contains all graphics and their different sizes.

For example, the count for `contentStats.graphics` can be `1` and the count of graphics object in embeds could be `2` (graphics object for normal screens, graphics object for mobile screens). 

Therefore, using that to count the number of sharable graphics can be inaccurate. 

As we do not have any identifier for graphics object for normal screen (which is the graphics size we are using for syndication) in the embeds array.

We can
1) Add a way to identify graphics objects for normal screens in the embeds array. That would require work on [next-es-interface](https://github.com/Financial-Times/next-es-interface/blob/master/server/models/embeds/index.js)
2) Get the graphics included in the content document, and look up the graphics object in the embeds array to check the `canBeSyndicated` property.